crates/astria-gossipnet/src/network.rs

```diff
@@ -131,43 +136,119 @@ impl Network {
 
         // build a gossipsub network behaviour
         let gossipsub = gossipsub::Behaviour::new(
-            gossipsub::MessageAuthenticity::Signed(local_key),
+            gossipsub::MessageAuthenticity::Signed(keypair),
             gossipsub_config,
         )
         .map_err(|e| eyre!("failed to create gossipsub behaviour: {}", e))?;
 
         let mut swarm = {
-            #[cfg(feature = "mdns")]
-            let mdns = mdns::tokio::Behaviour::new(mdns::Config::default(), local_peer_id)?;
+            let kademlia = if self.with_kademlia {
+                let mut cfg = KademliaConfig::default();
+                cfg.set_query_timeout(Duration::from_secs(5 * 60));
```

Comment:
Should the timeout be configurable? If not, please add this to the documentation, maybe on `fj with_kademlia`